docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,547 @@
+        },
+        "seed": {
+            "name": "<seed-name>",
+            "uid": "<seed-uid>",
+        },
+        "garden": {
+            "id": "<garden-cluster-identity>",
+        },
+    }
+}
+```
+
+### Distribution of Workload Identity Tokens
+
+Gardenlet will request tokens as per the global configurations and renew them
+regularly. It will be responsible to provide information for the specific usage
+of the token, e.g. shoot name, namespace and UID, via the `TokenRequest` API.
+`gardenlet` will be the only Gardener component authorized to create and refresh
+workload identity tokens. Seed Authorizer will be extended to allow gardenlets
+to request workload identity tokens only for `WorkloadIdentity` that they are
+responsible for.
+
+As the tokens will usually have lifetime shorter than the period between two
+reconciliations, it is essential that the token creation and management are
+decoupled from the current control loops of gardenlet and implemented by a
+dedicated controller, also running as part of the gardenlet. Forced renewal of
+the tokens will be performed when the resource referring `WorkloadIdentity`s is
+annotated with `gardener.cloud/operation=renew-workload-identity-token`. The
+annotation is deliberately not set on the `WorkloadIdentity` because single
+`WorkloadIdentity` can be used by multiple shoots potentially running on
+different seeds, i.e. multiple controllers would be responsible to react on the
+annotation which is usually fine, but all of them would have to negotiate when
+the operation is completed and the annotation to be removed.
+
+Kubernetes Secrets will be the resource holding the JWT, the provider config,
+and metadata about the used workload identity. The JWT will be stored under the
+`token` data key, while the provider config will use the `config` data key. The
+name and namespace of the used `WorkloadIdentity` will be stored in the
+annotations `workloadidentity.authentication.gardener.cloud/name` and
+`workloadidentity.authentication.gardener.cloud/namespace` respectively. The
+annotation `workloadidentity.authentication.gardener.cloud/context-object`, if
+present, will hold reference to the object using the workload identity, the
+value will be JSON document and have the format
+`{"apiVersion":"...","kind":"...","name":"...","namespace":"...","uid":"..."}`.
+It will be the source for the `spec.contextObject` field of the `TokenRequest`.
+To let the controller(s) easily select or distinguish these secrets, they will
+be labeled with
+`authentication.gardener.cloud/purpose: workload-identity-token-requestor`. The
+secrets will be also labeled with
+`workloadidentity.authentication.gardener.cloud/provider=<WorkloadIdentity.spec.targetSystem.type>`
+so that the extensions can easily select them and make adjustments via admission
+webhooks, e.g. transform the service provider config and the token into
+canonical form usable by the respective service provider SDK.
```

Comment:
Could you perhaps just write down an example YAML `Secret` similar to the one from the comments above? It is way easier to read this compared to this text paragraph.

Reply:
Done with 76f3c7d17